lib/src/Uncertainty/Algorithm/WeightedExperiments/openturns/SmolyakExperiment.hxx

```diff
@@ -0,0 +1,115 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Abstract top-level view of a Smolyak experiment
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_SMOLYAKEXPERIMENT_HXX
+#define OPENTURNS_SMOLYAKEXPERIMENT_HXX
+
+#include "openturns/WeightedExperimentImplementation.hxx"
+#include "openturns/WeightedExperiment.hxx"
+#include "openturns/Collection.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+
+
+/**
+ * @class SmolyakExperiment
+ *
+ * The class describes the probabilistic concept of Smolyak experiment
+ */
+class OT_API SmolyakExperiment
+  : public WeightedExperimentImplementation
+{
+  CLASSNAME
+public:
+
+  typedef Collection<WeightedExperiment> WeightedExperimentCollection;
+  typedef PersistentCollection<WeightedExperiment> WeightedExperimentPersistentCollection;
+
+  /** Compare two points approximately */
+  static bool comparePointsApproximately(const Point x, const Point y);
```

Comment:
should be private, uppercase

Reply:
I will use uppercase and refs. On the "private" topic, I have an issue here. 
- If I make the method private, then I cannot test it. This is an issue, because it is the main building block of the merging step. If this method is wrong, then the merging step will fail. Hence, testing it is important. 
- If the method is public, I can test it, which is done at C++ level in the `test_4` function of the unit test, at  [t_SmolyakExperiment_std.cxx](https://github.com/mbaudin47/openturns/blob/8318db107efffc32b46b7520f6810f8479c3c1ff/lib/test/t_SmolyakExperiment_std.cxx#L237). 
- On the other hand, the method is not the main point of the `SmolyakExperiment` class. From this point of view, it makes sense not to provide it.
- If there any design option which could make this method testable, still ?